app/Http/Controllers/Api/ReportbackController.php

```diff
@@ -23,15 +24,28 @@ public function __construct(ReportbackService $reportbackService)
     }
 
     /**
-     * Store a newly created resource in storage.
-     * @todo upsert reportbacks by default.
+     * Store a newly created resource in storage or
+     * update a reportback if it already exists.
+     *
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function create(Request $request)
+    public function store(ReportbackRequest $request)
     {
-        $reportback = $this->reportbackService->create($request->all());
+        // @TODO - instead should probably just have a method that gets northstar_id by default from a drupal_id if that is the only thing provided and then use that to find the reportback.
```

Comment:
I like the sound of that cause then you would never need the `type` to know the difference. It's always ends up as a `northstar_id`. But cool for a later thing :) Good to note it!